app/components/github/repo-sync.js

```diff
@@ -1,51 +1,85 @@
 import Component from '@ember/component';
 import { get, computed } from '@ember/object';
 import { alias } from '@ember/object/computed';
+import { task, timeout } from 'ember-concurrency';
 
-const TOTAL_STEPS = 7;
+const TOTAL_STEPS = 9;
 
 export default Component.extend({
   classNames: ['github__repo-sync'],
 
   githubRepo: null,
+  projectGithubRepo: null,
 
-  errored: computed('state', function() {
-    let state = get(this, 'state');
-    return state && state.indexOf('error') > -1;
+  pollServerForChanges: task(function* () {
+    let projectGithubRepo = get(this, 'projectGithubRepo');
+    let synced = get(projectGithubRepo, 'syncState') === 'synced';
+    let errored = get(this, 'errored');
+    while (!synced && !errored) {
+      yield timeout(2000);  // wait 2 seconds
+      get(this, 'githubRepo').reload();
+      projectGithubRepo.reload();
+    }
+  }).on('didRender').cancelOn('willDestroyElement').restartable(),
+
+  errored: computed('projectRepoState', 'repoState', function() {
+    let projectRepoState = get(this, 'projectRepoState');
+    let repoState = get(this, 'repoState');
+    return (repoState && repoState.indexOf('error') > -1)
+        || (projectRepoState && projectRepoState.indexOf('error') > -1);
   }),
 
   percentage: computed('stepNumber', function() {
     let stepNumber = get(this, 'stepNumber');
     return (stepNumber / TOTAL_STEPS) * 100;
   }),
 
-  state: alias('githubRepo.syncState'),
+  projectRepoState: alias('projectGithubRepo.syncState'),
+  repoState: alias('githubRepo.syncState'),
+
+  stepNumber: computed('projectRepoState', 'repoState', function() {
+    let projectRepoState = get(this, 'projectRepoState');
+    let repoState = get(this, 'repoState');
+
+    let state = projectRepoState;
+
+    if (!state
+        || state === 'unsynced'
+        || state === 'syncing_github_repo'
+        || state === 'errored_syncing_github_repo') {
+      state = repoState;
+    }
```

Comment:
I'm thinking rewrite this as

```Javascript
if (['syncing_tasks', 'syncing_comments', 'synced'].indexOf(state) == -1) {
  state = repoState;
}
```

Reply:
You can also extract to a separate function, something like `getRenderedState(projectRepoState, repoState)`, or it could be a computed. You can also do both.